crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -66,7 +67,8 @@ impl RoomExt for Room {
 pub(super) trait RoomDataProvider: Clone + Send + Sync + 'static {
     fn own_user_id(&self) -> &UserId;
     fn room_version(&self) -> RoomVersionId;
-    async fn profile(&self, user_id: &UserId) -> Option<Profile>;
+    async fn profile_from_user_id(&self, user_id: &UserId) -> Option<Profile>;
+    async fn profile_from_latest_event(&self, latest_event: &LatestEvent) -> Option<Profile>;
```

Comment:
Instead of two methods, could we have a single one that starts with trying to read from the latest event if the latest event's user id matches the one passed as a parameter, or the one from storage otherwise?

Reply:
That was my initial code! But I thought it was much cleaner to work like this. The function name and arguments are much simpler this way. Also, it moves the responsibility on the caller to know what to do if one fails, which I guess is a more desirable API.